ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/DropSourceCommand.java

```diff
@@ -58,16 +70,57 @@ public DdlCommandResult run(MetaStore metaStore, boolean isValidatePhase) {
         new DropTopicCommand(dataSource.getKsqlTopic().getTopicName());
     metaStore.deleteSource(sourceName);
     dropTopicCommand.run(metaStore, isValidatePhase);
-    if (!isValidatePhase && dataSource.getKsqlTopic().getKsqlTopicSerDe().getSerDe()
-                            == DataSource.DataSourceSerDe.AVRO) {
-      try {
-        schemaRegistryClient
-            .deleteSubject(sourceName + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX);
-      } catch (Exception e) {
-        throw new KsqlException("Could not clean up the schema registry for topic: " + sourceName,
-                                e);
+
+    deleteTopicIfNeeded(dataSource, isValidatePhase);
+
+    return new DdlCommandResult(true, "Source " + sourceName + " was dropped. "
+                                      + (withTopic ? "Topic '"
+                                                    + dataSource.getKsqlTopic().getTopicName()
+                                                    + "' was marked for deletion. Actual deletion "
+                                                    + "and removal from brokers may take some time "
+                                                    + "to complete." : ""));
+  }
+
+  private void deleteTopicIfNeeded(StructuredDataSource dataSource, boolean isValidatePhase) {
+    if (!isValidatePhase && withTopic) {
+      int retries = 0;
+      while (retries < NUM_RETRIES) {
```

Comment:
We can extract the retry logic into a method and use it in both places here, i.e, 

```
private void executeWithRetries(Callable<Void> callable, String errorMessage) {
    int retries = 0;
    while(retries < NUM_RETRIES) {
      try {
        callable.call();
      } catch (Exception e) {
        retries++;
      }
    }

    if (retries == NUM_RETRIES) {
      throw new KsqlException(errorMessage);
    }
}

Reply:
Made the change.